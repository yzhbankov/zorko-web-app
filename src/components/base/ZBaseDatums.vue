<template>
  <v-flex xs12>
    <v-list two-line subheader>
      <ZBaseDatumsItem
        v-for="item in items"
        :key="item.name"
        :name="item.name"
        :title="item.title"
        :connection="item.connection"
        :description="item.description"
        :info="item.info"
        :looksCount="item.looksCount"
        @datumActivated="onDatumActivated"
      />
    </v-list>
  </v-flex>
</template>

<script>
import ZBaseDatumsItem from './ZBaseDatumsItem.vue'

export default {
  name: 'ZBaseDatums',
  components: {
    ZBaseDatumsItem
  },
  props: {
    items: {
      type: Array
    }
  },
  methods: {
    onDatumActivated (item) {
      this.$emit('datumInput', item)
    }
  }
}
</script>
