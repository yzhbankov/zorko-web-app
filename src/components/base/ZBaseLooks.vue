<template>
  <v-container fluid v-bind="{ [`grid-list-${size}`]: true }">
    <v-layout row wrap>
      <v-flex
        v-for="item in items"
        :key="item.name"
        xs4>
        <ZBaseLooksItem
          :src="item.preview"
          :title="item.title"
          :path="item.path"
          :author="item.author"
          :datum="item.datum"
          @lookActivated="onLookActivated"
          @datumActivated="onDatumActivated"
        />
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
import ZBaseLooksItem from './ZBaseLooksItem.vue'

export default {
  name: 'ZBaseLooks',
  components: {
    ZBaseLooksItem
  },
  props: {
    items: {
      type: Array
    }
  },
  methods: {
    onLookActivated (item) {
      this.$emit('lookInput', item)

      this._routeTo(item.path)
    },
    onDatumActivated (item) {
      this.$emit('datumInput', item)

      this._routeTo('/datums/' + item.name)
    },
    _routeTo (path) {
      this.$router && this.$router.push(path)
    }
  },
  data () {
    return {
      size: 'lg'
    }
  }
}
</script>
