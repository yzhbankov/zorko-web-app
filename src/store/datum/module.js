import state from './state'
import * as mutations from './mutations'
import * as actions from './actions'

export default ({namespaced}) => ({
  namespaced,
  state: state(),
  mutations,
  actions
})
