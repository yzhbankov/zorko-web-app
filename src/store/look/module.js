import state from './state'
import * as actions from './actions'
import * as mutations from './mutations'

export default ({namespaced}) => ({
  namespaced,
  state: state(),
  actions,
  mutations
})
