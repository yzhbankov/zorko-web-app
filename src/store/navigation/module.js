import state from './state'
import * as mutations from './mutations'

export default ({namespaced}) => ({
  namespaced,
  state: state(),
  mutations
})
