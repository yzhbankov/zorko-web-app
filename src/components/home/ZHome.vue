<template>
    <v-layout>
      <v-flex xs12>
          <ZBaseLooks
            v-if="looksPreview"
            :items="popularLooks"
            @lookInput="onLookInput"
            @datumInput="onOpenDatum"/>
          <ZBaseDatums
            v-if="datumsPreview"
            :items="popularDatums"
            @datumInput="onOpenDatum"
          />
      </v-flex>
    </v-layout>
</template>

<script>
import ZBaseLooks from '../base/ZBaseLooks.vue'
import ZBaseDatums from '../base/ZBaseDatums.vue'

import { createNamespacedHelpers } from 'vuex'
const { mapGetters } = createNamespacedHelpers('home/')

export default {
  name: 'Home',
  components: {
    ZBaseLooks,
    ZBaseDatums
  },
  props: {
    looksPreview: {
      type: Boolean
    },
    datumsPreview: {
      type: Boolean
    }
  },
  computed: {
    ...mapGetters([
      'popularLooks',
      'popularDatums'
    ])
  },
  methods: {
    onLookInput (item) {
      this.$router.push(item.path)
    },
    onOpenDatum ({name}) {
      this.$router.push('/datums/' + name)
    }
  }
}
</script>
