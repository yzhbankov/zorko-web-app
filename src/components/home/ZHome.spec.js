import {createLocalVue, shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import Vuex from 'vuex'
import ZHome from './ZHome.vue'
import ZBaseLooks from '../base/ZBaseLooks.vue'
import ZBaseDatums from '../base/ZBaseDatums.vue'
import module from '../../store/home/module'

import mockLooks from '../../../static/__mocks__/looks.json'
import mockDatums from '../../../static/__mocks__/datums.json'

const localVue = createLocalVue()
const homeModule = module({namespaced: true})

localVue.use(Vuex)

describe('ZHome.vue', () => {
  let store
  let state
  let propsData

  beforeEach(() => {
    state = {
      ...homeModule.state,
      looks: mockLooks,
      datums: mockDatums
    }

    propsData = {
      looksPreview: false,
      datumsPreview: false
    }

    store = new Vuex.Store({
      state: {},
      modules: {
        home: {
          ...homeModule,
          state
        }
      }
    })
  })

  it('passes props to `ZBaseLooks`', () => {
    propsData.looksPreview = true
    const wrapper = shallow(ZHome, {
      store,
      localVue,
      propsData
    })

    const {vm} = wrapper.find(ZBaseLooks)

    expect(vm.$props.items).toEqual(mockLooks)
  })

  it('passes props to `ZBaseDatums`', () => {
    propsData.datumsPreview = true
    const wrapper = shallow(ZHome, {
      store,
      localVue,
      propsData
    })

    const {vm} = wrapper.find(ZBaseDatums)

    expect(vm.$props.items).toEqual(mockDatums)
  })

  it('changes route because datum`s item `openDatum` event', () => {
    propsData.datumsPreview = true
    const datumInfo = {name: 'bazzz'}
    const $router = {
      push: jest.fn()
    }
    const wrapper = shallow(ZHome, {
      store,
      localVue,
      propsData,
      mocks: {
        $router
      }
    })
    const {vm} = wrapper.find(ZBaseDatums)

    vm.$emit('datumInput', datumInfo)

    expect($router.push).toHaveBeenCalledWith('/datums/bazzz')
  })

  it('has same HTML for looks preview', () => {
    const renderer = createRenderer()
    const homeWrapper = shallow(ZHome, {
      store,
      localVue,
      propsData
    })

    renderer.renderToString(homeWrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
