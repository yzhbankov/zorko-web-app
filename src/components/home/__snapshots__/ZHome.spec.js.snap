// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZHome.vue has same HTML for looks preview 1`] = `
<v-layout>
    <v-flex xs12="">
        <!---->
        <!---->
    </v-flex>
</v-layout>
`;
